sdk/core/azure-core/azure/core/pipeline/policies/_retry.py

```diff
@@ -84,6 +84,7 @@ class RetryPolicy(HTTPPolicy):
 
     #: Maximum backoff time.
     BACKOFF_MAX = 120
+    _RETRY_CODES = [i for i in range(999) if i not in [i for i in range(500) if i != 408] + [501, 505]]
```

Comment:
Does this need to be a class attr? 

Also FWIW I think the a construction more similar to the previous one is more clear:
```
SAFE_CODES = set(range(506)) - set([408, 500, 502, 503, 504])
RETRY_CODES = set(range(999)) - SAFE_CODES
```
but then this line: 
```
self._retry_on_status_codes = set(status_codes + retry_codes)
```
would need to change:
```
self._retry_on_status_codes = set(status_codes).union(RETRY_CODES)
```
(and possibly other places)

Reply:
Also, as long as we are optimizing, constructing with sets is ~1000x faster:
```
In [3]: %timeit SAFE_CODES = set(range(506)) - set([408, 500, 502, 503, 504]); RETRY_CODES = set(range(999)) - SAFE_CODES
10000 loops, best of 3: 59.8 µs per loop

In [4]: %timeit _RETRY_CODES = [i for i in range(999) if i not in [i for i in range(500) if i != 408] + [501, 505]]
10 loops, best of 3: 83.9 ms per loop
```

Reply:
good suggestion (y) Changed to use sets.